### PR TITLE
refactor server update behavior

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
refactor server update behavior

## Motivation and Context
add support for `minisign` signature verification during updates,
this is optional and can also be controlled from air-gapped location.

## How to test this PR?
To test this locally build binary locally with this PR

```
~ GO111MODULE=on CGO_ENABLED=0 go build -tags kqueue -trimpath --ldflags "-s -w -X github.com/minio/minio/cmd.Version=2020-07-21T01:08:20Z -X github.com/minio/minio/cmd.ReleaseTag=DEVELOPMENT.2020-07-21T01-08-20Z -X github.com/minio/minio/cmd.CommitID=da62d7ea0b9829c0ad9fba59a53c47897caff338 -X github.com/minio/minio/cmd.ShortCommitID=da62d7ea0b98 -X github.com/minio/minio/cmd.GOPATH=/home/harsha/mygo -X github.com/minio/minio/cmd.GOROOT=/home/harsha/.gimme/versions/go1.13.13.linux.amd64" -o /home/harsha/mygo/src/github.com/minio/minio/minio 1>/dev/null
~ for i in {02..09}; do cp -a ./minio $(which minio)${i}; done
```

```sh
#!/usr/bin/env bash

export MINIO_PROMETHEUS_AUTH_TYPE=public
export MINIO_ACCESS_KEY="minio"
export MINIO_SECRET_KEY="minio123"
export MINIO_UPDATE_MINISIGN_PUBKEY="RWTx5Zr1tiHQLwG9keckT0c45M3AGeHD6IvimQHpyRywVWGbP1aVSGav"

endpoints=""

for i in {02..09}; do
    endpoints="$endpoints https://localhost:90${i}/tmp/${i}"
done

for i in {02..09}; do
    minio${i} server --certs-dir ../minio-go/testcerts/ --address ":90${i}" ${endpoints} &
done
```

```
~ mc admin update myminio-ssl
```

Would validate the server checksum and signature.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
To test this locally build binary locally with this PR

```
~ GO111MODULE=on CGO_ENABLED=0 go build -tags kqueue -trimpath --ldflags "-s -w -X github.com/minio/minio/cmd.Version=2020-07-21T01:08:20Z -X github.com/minio/minio/cmd.ReleaseTag=DEVELOPMENT.2020-07-21T01-08-20Z -X github.com/minio/minio/cmd.CommitID=da62d7ea0b9829c0ad9fba59a53c47897caff338 -X github.com/minio/minio/cmd.ShortCommitID=da62d7ea0b98 -X github.com/minio/minio/cmd.GOPATH=/home/harsha/mygo -X github.com/minio/minio/cmd.GOROOT=/home/harsha/.gimme/versions/go1.13.13.linux.amd64" -o /home/harsha/mygo/src/github.com/minio/minio/minio 1>/dev/null
~ for i in {02..09}; do cp -a ./minio $(which minio)${i}; done
```

```sh
#!/usr/bin/env bash

unset MINIO_KMS_KES_CERT_FILE
unset MINIO_KMS_KES_KEY_FILE
unset MINIO_KMS_KES_ENDPOINT
unset MINIO_KMS_KES_KEY_NAME

export MINIO_PROMETHEUS_AUTH_TYPE=public
export MINIO_ACCESS_KEY="minio"
export MINIO_SECRET_KEY="minio123"
export MINIO_UPDATE_MINISIGN_PASSWD="RWTx5Zr1tiHQLwG9keckT0c45M3AGeHD6IvimQHpyRywVWGbP1aVSGav"
export MINIO_KMS_MASTER_KEY=my-minio-key:6368616e676520746869732070617373776f726420746f206120736563726574

endpoints=""

for i in {02..09}; do
    endpoints="$endpoints https://localhost:90${i}/tmp/${i}"
done

for i in {02..09}; do
    minio${i} server --certs-dir ../minio-go/testcerts/ --address ":90${i}" ${endpoints} &
done
```

```
~ mc admin update myminio-ssl
```

Would validate the server checksum and signature.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-zoned.sh | :x: [more...](#0fe401eb-5407-4ac3-8bc4-01c1820e4bde) |
<a name="0fe401eb-5407-4ac3-8bc4-01c1820e4bde"></a>
### 10107-6a67bee/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:30448
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 98989c52da3a:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 5 seconds
(5/15) Running awscli tests ... done in 1 minutes and 22 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 30 seconds
(8/15) Running minio-dotnet tests ... done in 47 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 43 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 25 seconds
(11/15) Running minio-js tests ... done in 1 minutes and 2 seconds
(12/15) Running minio-py tests ... done in 2 minutes and 32 seconds
(13/15) Running s3cmd tests ... FAILED in 19 seconds
{
  "name": "s3cmd",
  "duration": "8464",
  "function": "test_sync_list_objects",
  "status": "FAIL",
  "error": "WARNING: Bucket is not empty. Removing all the objects from it first. This may take some time...\nERROR: S3 error: 404 (Not Found)"
}
(13/15) Running s3select tests ... done in 8 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#d4645a06-e673-4c17-b0e0-093cc40a12e9) |
| mint-zoned.sh | :x: [more...](#e2a3576e-0681-4328-9068-a1e5a394518f) |
<a name="e2a3576e-0681-4328-9068-a1e5a394518f"></a>
### 10107-fef26bf/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:32361
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1c058ec980ec:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 2 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 44 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 1 minutes and 24 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 38 seconds
(8/15) Running minio-dotnet tests ... done in 56 seconds
(9/15) Running minio-go tests ... done in 1 minutes and 53 seconds
(10/15) Running minio-java tests ... done in 1 minutes and 29 seconds
(11/15) Running minio-js tests ... done in 49 seconds
(12/15) Running minio-py tests ... done in 2 minutes and 41 seconds
(13/15) Running s3cmd tests ... FAILED in 22 seconds
{
  "name": "s3cmd",
  "duration": "12179",
  "function": "test_sync_list_objects",
  "status": "FAIL",
  "error": "WARNING: Bucket is not empty. Removing all the objects from it first. This may take some time...\nERROR: S3 error: 404 (Not Found)"
}
(13/15) Running s3select tests ... done in 8 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="d4645a06-e673-4c17-b0e0-093cc40a12e9"></a>
### 10107-fef26bf/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:32725
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dff2d5453bc9:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 5 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... FAILED in 1 minutes and 12 seconds
{
  "name": "awscli",
  "duration": 1557,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}
(5/15) Running healthcheck tests ... done in 1 seconds
(6/15) Running mc tests ... done in 3 minutes and 16 seconds
(7/15) Running minio-dotnet tests ... done in 1 minutes and 34 seconds
(8/15) Running minio-go tests ... done in 5 minutes and 35 seconds
(9/15) Running minio-java tests ... done in 6 minutes and 19 seconds
(10/15) Running minio-js tests ... done in 2 minutes and 34 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 58 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 23 seconds
(13/15) Running s3select tests ... done in 1 minutes and 5 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: cmd/common-main.go [hunk 1]
```diff
@@ -68,16 +70,43 @@ func verifyObjectLayerFeatures(name string, objAPI ObjectLayer) {
...
-	if updateMsg, _, currentReleaseTime, latestReleaseTime, err := getUpdateInfo(2*time.Second, mode); err == nil {
-		if updateMsg == "" {
-			return
-		}
-		if globalInplaceUpdateDisabled {
-			logStartupMessage(updateMsg)
-		} else {
-			logStartupMessage(prepareUpdateMessage("Run `mc admin update`", latestReleaseTime.Sub(currentReleaseTime)))
-		}
+	crTime, err := GetCurrentReleaseTime()
```
Comment (nitisht) [inline cmd/common-main.go:84]: Comment seems out of place

**Review by nitisht — APPROVED** [review]
LGTM & Tested, minor nit

File: cmd/common-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/common-main.go:84]: It says why we return without returning error.

**Pull request merged**